### PR TITLE
AirfRANS: lr=2e-4+T_max=10 — even lower LR for deeper basin

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

lr=3e-4 beat lr=5e-4 for AirfRANS (0.0197 vs 0.0207). Does the trend continue? lr=2e-4 may find an even deeper phase transition basin at the cost of slower convergence. The progression 5e-4→3e-4→better suggests the AirfRANS loss landscape rewards very conservative learning rates for escaping into deep basins. lr=2e-4 is the natural next step in this sweep.

## Instructions

Run exactly this command:
```bash
cd target/icml2026 && python train.py --dataset airfrans --airfrans_task full --optimizer adamw --lr 2e-4 --cosine_t_max 10 --no-use-ema --enable-fourier --epochs 999 --wandb_group airfrans-lr-sweep
```

The model will run until the SENPAI timeout (~30 min or ~41 epochs based on prior runs). Report `val_primary/surface_mse` at the best epoch. Pay special attention to:
1. Whether the phase transition still occurs (it may be delayed vs lr=3e-4)
2. If it occurs, what epoch it happens at and what depth it reaches
3. If no phase transition within the time budget, note the final val loss trajectory

## Baseline

- **Dataset:** AirfRANS (full task)
- **Primary metric:** `val_primary/surface_mse` (lower is better)
- **Current best:** 0.0197 (PR #2614, gilbert, lr=3e-4, T_max=10, 41 epochs, W&B run: v5ka7832)
- **lr=5e-4 best:** 0.0207 (PR #2617, T_max=10, 41 epochs, W&B run: z7t3ibwi)
- **External target:** 0.0043 (~4.6x gap remaining)
- **Reproduce baseline:** `cd target/icml2026 && python train.py --dataset airfrans --airfrans_task full --optimizer adamw --lr 3e-4 --cosine_t_max 10 --no-use-ema --enable-fourier --epochs 999`

LR sweep context: 5e-4 → 0.0207, 3e-4 → 0.0197. Testing if 2e-4 continues the trend downward.

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT fern:

## Results

### Primary Metric

| Metric | Baseline (lr=3e-4) | This Run (lr=2e-4, best) | Delta |
|--------|--------------------|-----------------------|-------|
| val_primary/surface_mse | 0.0197 | 0.0306 (epoch 31) | +55.3% |
| test_primary/surface_mse | — | 0.1020 (epoch 41*) | N/A* |

*Test evaluated at final epoch 41 (restart spike), not best val epoch. Not representative of best model state.

### Phase Transition Trajectory

| Epoch | val_primary/surface_mse | surface_mse_p | Phase |
|-------|------------------------|---------------|-------|
| 1 | 0.6817 | 2.674 | Pre-transition |
| 9 | 0.2367 | 0.946 | First dip below old baseline |
| 15 | 0.2185 | 0.874 | Gradual descent |
| 19 | 0.1679 | 0.671 | **Transition onset** |
| 21 | 0.1017 | 0.406 | Deep transition |
| 24 | 0.0637 | 0.254 | Accelerating |
| 29 | 0.0358 | 0.143 | Near-best |
| **31** | **0.0306** | **0.122** | **Best** |
| 33 | 0.0388 | 0.155 | Post-best oscillation |
| 38 | 0.0389 | 0.155 | Not recovering |
| 41 | 0.0955 | 0.382 | Restart spike (test eval) |

### Experiment Details

- **Command:** `cd target/icml2026 && python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 2e-4 --cosine-t-max 10 --no-use-ema --enable-fourier --epochs 999 --agent fern --wandb-name "fern/airfrans-lr2e4-tmax10" --wandb-group airfrans-lr-sweep`
- **W&B Run:** `6hl0j2kn`
- **Epochs completed:** 41 (hit 30-min timeout)
- **Peak memory:** Not recorded

### What happened

**The phase transition does occur at lr=2e-4, but it's delayed and shallower than lr=3e-4 within the same epoch budget.**

1. **Phase transition confirmed:** It begins around epoch 19 (0.168) and reaches its deepest at epoch 31 (0.0306). This is delayed vs lr=3e-4 which reached 0.0197 in the same 41 epochs.

2. **Wider oscillations:** The T_max=10 cosine restart creates extreme oscillations at depth — the model bounces between 0.03 and 0.18 in the deep basin. At lr=3e-4, these oscillations may be tighter, allowing the model to consolidate gains more consistently.

3. **Post-transition stall:** After epoch 31 (0.0306), subsequent trough epochs didn't beat it (epoch 33: 0.0388, epoch 38: 0.0389). The model appears to have reached a depth limit for lr=2e-4 within this training budget, whereas lr=3e-4 pushed through to 0.0197.

4. **The LR sweep trend reverses:** 5e-4 → 0.0207, 3e-4 → 0.0197, 2e-4 → 0.0306. The trend of "lower LR = deeper basin" does NOT continue. lr=3e-4 is the sweet spot — lr=2e-4 is too conservative, causing slower convergence and wider restart oscillations that prevent the model from reaching the same depth within the time budget.

### Suggested follow-ups

1. **lr=2.5e-4:** If there's a sweet spot between 2e-4 and 3e-4, it could be worth testing.
2. **lr=2e-4 with T_max=20 or 30:** Wider cosine cycles might reduce the extreme oscillation amplitude at depth.
3. **Longer training for lr=2e-4:** The model stalled post-epoch 31 but with 2-3x more epochs, it might break through. However, this seems unlikely to beat lr=3e-4 which was already better at the same epoch count.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2657 because ADVISOR: Dead end — AirfRANS lr=2e-4 reached 0.0306 (vs baseline 0.01841). The LR lower bound is now confirmed: lr=2e-4 is too conservative, causing delayed/shallow phase transition. LR sweet spot is 3e-4 to 7e-4, with 7e-4 currently optimal. Good trajectory analysis, fern. Closing..
